src/util/FetchUtil.ts

```diff
@@ -1,29 +1,63 @@
-import fetch from '@rdfjs/fetch';
-import type { DatasetResponse } from '@rdfjs/fetch-lite';
-import type { Dataset } from 'rdf-js';
+import type { Response } from 'cross-fetch';
+import { fetch } from 'cross-fetch';
+import { Store } from 'n3';
+import { BasicRepresentation } from '../ldp/representation/BasicRepresentation';
 import { getLoggerFor } from '../logging/LogUtil';
-import { createErrorMessage } from './errors/ErrorUtil';
+import type { RepresentationConverter } from '../storage/conversion/RepresentationConverter';
+import { INTERNAL_QUADS } from './ContentTypes';
+import { BadRequestHttpError } from './errors/BadRequestHttpError';
 
 const logger = getLoggerFor('FetchUtil');
 
 /**
  * Fetches an RDF dataset from the given URL.
+ * Input can also be a Response if the request was already made.
+ * In case the given Response object was already parsed its body can be passed along as a string.
+ *
+ * The converter will be used to convert the response body to RDF.
  */
-export async function fetchDataset(url: string): Promise<Dataset> {
-  let rawResponse: DatasetResponse<Dataset>;
-  try {
-    rawResponse = (await fetch(url)) as DatasetResponse<Dataset>;
-  } catch (err: unknown) {
-    logger.error(`Cannot fetch ${url}: ${createErrorMessage(err)}`);
-    throw new Error(`Cannot fetch ${url}`);
+export async function fetchDataset(url: string, converter: RepresentationConverter): Promise<Store>;
+export async function fetchDataset(response: Response, converter: RepresentationConverter, body?: string):
+Promise<Store>;
+export async function fetchDataset(input: string | Response, converter: RepresentationConverter, body?: string):
```

Comment:
I'd split this in 2 methods; one that fetches the stream, the other that calls that method and does the rest.

Then we can use the stream separately (see above).

Reply:
What stream would you expect from the first one? There are several streams involved here. I think returning the Representation might make sense? That is a stream of Quad objects then.

Reply:
Yeah, quads!